### DISCUSSION TITLE
Is MAUI Blazor Hybrid supported ?

### DISCUSSION THREAD

**Discussion by orosbogdan** [LostBeard/SpawnDev.BlazorJS#40] [Q&A]
Is this library compatible with MAUI Blazor ?

**Comment by LostBeard** [comment]
Short answer:  
Yes, but only in Blazor WebAssembly. 

I haven't tried MAUI in a few months. Looking at the VS templates `.NET MAUI Blazor Hybrid and Web App` looked like the best bet so I created a new project with it.

It creates 4 projects, with 2 main projects.

With a project named "MauiWebHybrid" the projects names created are:  
- MauiWebHybrid - Native application. Platforms: Android, iOS, MacCatalyst, Tizen, Windows
- MauiWebHybrid.Shared - Shared reference. Referenced by MauiWebHybrid and MauiWebHybrid.Web.Client
- MauiWebHybrid.Web - Blazor United Server application. Platform: web
- MauiWebHybrid.Web.Client - Blazor WebAssembly client. Referenced by the Blazor Server (MauiWebHybrid.Web)

SpawnDev.BlazorJS can be used in the MauiWebHybrid.Web.Client Blazor WASM project. Prerendering has to be disabled for components that use BlazorJS or globally ([more info](https://github.com/LostBeard/SpawnDev.BlazorJS?tab=readme-ov-file#compatible-blazor-web-app-options)).

But the Blazor WASM project will only be used for a platform web build.

It does seem to be possible to run Blazor WASM on all of the platforms. Build a "Blazor WebAssembly Standalone App" and place the resulting index.html and other build files in the MauiWebHybrid project's wwwroot folder, replacing the index.html file in there.

Then, when you run a debug build for Android (for example) it will load a webview like it normally does but instead of loading the MAUI Blazor Javascript library it loads the Blazor WASM app. I haven't had time yet to figure out how to access native APIs from Blazor WASM in the webview yet though.

**Comment by LostBeard** [comment]
I may have been able to get SpawnDev.BlazorJS working in the Shared project, but the `JSRuntime` used by the native platforms is `WebViewJSRuntime` and it does not appear to support synchronous calls to Javascript.
